rust/src/applayer.rs

```diff
@@ -535,14 +535,17 @@ pub trait Transaction {
 }
 
 pub trait State<Tx: Transaction> {
-    fn get_transactions(&self) -> &[Tx];
+    /// Return the number of transactions in the state's transaction collection.
+    fn get_transaction_count(&self) -> usize;
```

Comment:
in the C API we have a `StateGetTxCnt` callback (which is misnamed. It should be `StateGetTxMaxId`). How does it relate to this function/trait logic?